### PR TITLE
Improve performance in searchstream

### PR DESCRIPTION
**Pull request by Uzlopak**
Unit tests are green, so I think the change provides an equivalent change. 

Screenshot from 2021-11-29 19-10-02

Benchmark: 
```
> @fastify/busboy@1.0.0 bench:dicer
> node deps/dicer/bench/dicer-bench-multipart-parser.js

9090.91 mb/sec
11111.11 mb/sec
10000.00 mb/sec
11111.11 mb/sec
11111.11 mb/sec
11111.11 mb/sec
12500.00 mb/sec
12500.00 mb/sec
11111.11 mb/sec
11111.11 mb/sec
```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
Also deoptimization traces are gone. 

`node --trace-deopt --trace-opt ./deps/dicer/bench/dicer-bench-multipart-parser.js `

File: deps/streamsearch/sbmh.js [hunk 1]
```diff
@@ -101,18 +101,20 @@ SBMH.prototype._sbmh_feed = function(data) {
     while (pos < 0 && pos <= len - needle_len) {
```
Comment (Uzlopak) [inline deps/streamsearch/sbmh.js:101]: this while loop is now the biggest performance bottle neck. 

Maybe we optimize it some other day if it is possible.

**Comment by Uzlopak** [comment]
I think this PR is actually good enough performance gain for this iteration, so that even @mcollina  would be happy :-D as it does not only shaves some yoctoseconds. 

So we improved the performance from about 3,5 GB/s to  10-12 GB/s.

**Comment by kibertoad** [comment]
will take a look later tonight, need to wrap up version release at work

**Review by mcollina — APPROVED** [review]
lgtm

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
@Uzlopak BTW, I've published a beta version of the package and tried running fastify-multipart as well as multer v2 tests against it, and all of them are still passing, so looks like your optimizations didn't break anything :)

For the first public release I still want to wait for https://github.com/fastify/busboy/issues/26
Is there anything else that is on your "must have" list for 1.0.0?

**Comment by Uzlopak** [comment]
@kibertoad 
I think we already achieved alot :). Just let us merge the last PRs and then release it :)

**Pull request merged**